draft: fix: remove pointer receiver

rhnvrm opened the pull request:
Bench stat:

```
name                      old time/op    new time/op    delta
NoField-8                    179ns ± 3%     114ns ± 3%  -36.06%  (p=0.000 n=3+3)
OneField-8                   283ns ± 1%     244ns ± 3%  -13.69%  (p=0.009 n=3+3)
ThreeFields-8                338ns ± 2%     288ns ± 1%  -14.69%  (p=0.002 n=3+3)
ErrorField-8                 381ns ±13%     267ns ± 1%  -29.87%  (p=0.047 n=3+3)
HugePayload-8               1.11µs ± 6%    0.95µs ±21%     ~     (p=0.239 n=3+3)
ThreeFields_WithCaller-8     932ns ± 4%     405ns ± 2%  -56.57%  (p=0.001 n=3+3)
NoField_WithColor-8          320ns ± 1%     170ns ± 2%  -46.97%  (p=0.000 n=3+3)
OneField_WithColor-8         441ns ± 9%     323ns ± 3%  -26.79%  (p=0.023 n=3+3)
ThreeFields_WithColor-8      569ns ±19%     388ns ± 2%     ~     (p=0.081 n=3+3)
ErrorField_WithColor-8       500ns ±10%     350ns ± 2%  -30.00%  (p=0.027 n=3+3)
HugePayload_WithColor-8     1.24µs ± 7%    1.03µs ± 1%  -16.95%  (p=0.036 n=3+3)

name                      old alloc/op   new alloc/op   delta
NoField-8                    0.00B          0.00B          ~     (zero variance)
OneField-8                    336B ± 0%      336B ± 0%     ~     (zero variance)
ThreeFields-8                 336B ± 0%      336B ± 0%     ~     (zero variance)
ErrorField-8                  368B ± 0%      352B ± 0%     ~     (zero variance)
HugePayload-8                 919B ± 0%      919B ± 0%     ~     (zero variance)
ThreeFields_WithCaller-8      656B ± 0%      336B ± 0%     ~     (zero variance)
NoField_WithColor-8          0.00B          0.00B          ~     (zero variance)
OneField_WithColor-8          336B ± 0%      336B ± 0%     ~     (zero variance)
ThreeFields_WithColor-8       336B ± 0%      336B ± 0%     ~     (zero variance)
ErrorField_WithColor-8        368B ± 0%      352B ± 0%     ~     (zero variance)
HugePayload_WithColor-8       919B ± 0%      918B ± 0%     ~     (zero variance)

name                      old allocs/op  new allocs/op  delta
NoField-8                     0.00           0.00          ~     (zero variance)
OneField-8                    2.00 ± 0%      2.00 ± 0%     ~     (zero variance)
ThreeFields-8                 2.00 ± 0%      2.00 ± 0%     ~     (zero variance)
ErrorField-8                  4.00 ± 0%      3.00 ± 0%     ~     (zero variance)
HugePayload-8                 3.00 ± 0%      3.00 ± 0%     ~     (zero variance)
ThreeFields_WithCaller-8      7.00 ± 0%      2.00 ± 0%     ~     (zero variance)
NoField_WithColor-8           0.00           0.00          ~     (zero variance)
OneField_WithColor-8          2.00 ± 0%      2.00 ± 0%     ~     (zero variance)
ThreeFields_WithColor-8       2.00 ± 0%      2.00 ± 0%     ~     (zero variance)
ErrorField_WithColor-8        4.00 ± 0%      3.00 ± 0%     ~     (zero variance)
HugePayload_WithColor-8       3.00 ± 0%      3.00 ± 0%     ~     (zero variance)
```

mr-karan commented:
LGTM, we can merge this. 

Since this already has the strings.IndexFunc, I guess we can close https://github.com/zerodha/logf/pull/6

rhnvrm commented:
I have rebased and updated the commit again, please review. Not sure why the tests are failing exactly. Also hope it does not add any other un-intended consequence because of removing the pointer receiver and the tests cover most of the usecases. :crossed_fingers:
